ztest/ztest.go

```diff
@@ -0,0 +1,219 @@
+// Package ztest runs formulaic tests ("ztests") that apply a ZQL query to an
+// input and check for an expected output.
+//
+// A ztest is defined in a YAML file.
+//
+//    zql: count()
+//
+//    input: |
+//      #0:record[i:int64]
+//      0:[1;]
+//      0:[2;]
+//
+//    output: |
+//      #0:record[count:uint64]
+//      0:[2;]
+//
+// Input format is detected automatically and can be anything recognized by
+// "zq -i auto" (including optional gzip compression).  Output format defaults
+// to zng but can be set to anything accepted by "zq -f".
+//
+//    zql: count()
+//
+//    input: |
+//      #0:record[i:int64]
+//      0:[1;]
+//      0:[2;]
+//
+//    output-format: table
+//
+//    output: |
+//      COUNT
+//      2
+//
+// Ztest YAML files for a package should reside in a subdirectory named
+// testdata/ztest.  The package should also contain a file named ztest_test.go.
```

Comment:
Why is the recommendation to have a separate `ztest_test.go` file just to call `ztest.Run`?

The `filter` package has two files currently, `filter.go` and `filter_test.go`. Why shouldn't we add the example `func TestZTest(...)` to the existing `filter_test.go` file?

Reply:
Good question. It highlights that a package contains ztests since the YAML files themselves are somewhat hidden in testdata/ztest/. I’m not certain it’s worth the overhead, though. If you agree, I’ll replace it with guidance that TestZTest for pkg should go into pkg_test.go.

Reply:
👍